src/lib/sfge/SfgePathlessEngine.ts

```diff
@@ -1,8 +1,64 @@
+import {SfdxError, Messages} from '@salesforce/core';
 import {AbstractSfgeEngine, SfgeViolation} from "./AbstractSfgeEngine";
-import {RuleViolation} from '../../types';
-import {RuleType} from '../../Constants';
+import {Rule, RuleGroup, RuleTarget, RuleViolation, SfgeConfig} from '../../types';
+import {CUSTOM_CONFIG, MissingOptionsBehavior, RuleType} from '../../Constants';
+import {uxEvents, EVENTS} from '../ScannerEvents';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'SfgeEngine');
 
 export class SfgePathlessEngine extends AbstractSfgeEngine {
+	private missingOptionsBehavior: MissingOptionsBehavior;
+
+	/**
+	 * Invokes sync/async initialization required for the engine.
+	 * @override
+	 */
+	public async init(): Promise<void> {
+		if (this.initialized) {
+			return;
+		}
+		await super.init();
+		this.missingOptionsBehavior = await this.config.getMissingOptionsBehavior(SfgePathlessEngine.ENGINE_ENUM);
+		this.initialized = true;
+	}
+
+	/**
+	 * Helps make decision to run an engine or not based on the Rules, Target paths, and the Engine Options selected per
+	 * run. At this point, filtering should have already happened.
+	 * @override
+	 */
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public shouldEngineRun(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string,string>): boolean {
```

Comment:
Do you think this logic is reusable if we pass the engine as a parameter? If so, can we move this to a reusable place? The halt/warn/skip logic sounds helpful for any new engine we add in future.

Reply:
Good idea. We could conceivably generalize this and add it to `CommonEngineUtils` or something. Stand by.